app/src/main/java/com/example/androidacademyhomework/UI/MovieListFragment.kt

```diff
@@ -47,25 +48,45 @@ class MovieListFragment : Fragment() {
         }
     }
 
-    private val clickListener = object : OnRecyclerItemClicked {
-        override fun onClick(movie:Movie) {
-            goToMovieDetailFragment(movie.id)
-        }
-    }
-
-    private fun updateData() {
-        (recyclerMovieList?.adapter as? MovieAdapter)?.apply {
-            bindActors(MoviesDataSource().getMovies())
+    private fun updateDate() {
+        scope.launch {
+            var listMovies = listOf<Movie>()
+            whenStarted {
+                listMovies = loadMovies(requireContext())
+            }
+            if (listMovies.isNotEmpty()) {
+                (recyclerMovieList?.adapter as? MovieAdapter)?.apply {
+                    bindMovies(listMovies)
+                }
+            } else {
+                Toast.makeText(requireContext(), "Error load movies", Toast.LENGTH_SHORT).show()
```

Comment:
Can you access UI from background thread?

Reply:
No, but here, everything that is not marked as being in the background thread is done in the main. I think so.

Reply:
My mistake - you're launching on the lifecycle scope